unit/src/expr/require_expr.h

```diff
@@ -0,0 +1,33 @@
+/*******************************************************************\
+
+ Module: Unit test utilities
+
+ Author: DiffBlue Limited. All rights reserved.
+
+\*******************************************************************/
+
+/// \file
+/// Helper functions for requiring specific expressions
+/// If the expression is of the wrong type, throw a CATCH exception
+/// Also checks associated properties and returns a casted version of the
+/// expression.
+
+#ifndef CPROVER_SRC_EXPR_REQUIRE_EXPR_H
+#define CPROVER_SRC_EXPR_REQUIRE_EXPR_H
+
+#include <util/std_expr.h>
+
+class require_exprt
```

Comment:
@peterschrammel would it be OK as suggested by @reuk for me to make this into a `namespace`, I know they are forbidden by coding standard but I don't really understand why.